solr/contrib/ltr/src/java/org/apache/solr/ltr/LTRRescorer.java

```diff
@@ -181,14 +185,19 @@ public void scoreFeatures(IndexSearcher indexSearcher,
         readerContext = leaves.get(readerUpto);
         endDoc = readerContext.docBase + readerContext.reader().maxDoc();
       }
-      // We advanced to another segment
-      if (readerContext != null) {
-        docBase = readerContext.docBase;
-        scorer = modelWeight.scorer(readerContext);
+      try{
+        // We advanced to another segment
+        if (readerContext != null) {
+          docBase = readerContext.docBase;
+          scorer = modelWeight.scorer(readerContext);
+        }
+        scoreSingleHit(indexSearcher, topN, modelWeight, docBase, hitUpto, hit, docID, scoringQuery, scorer, reranked);
+        hitUpto++;
+      } catch (ExitableDirectoryReader.ExitingReaderException ex) {
+        break;
```

Comment:
I was reasoning about this, I think I prefer an all or nothing approach:
Try to rerank up to the topK I am targeting, in the allowed time, if I fail to succeed in the allowed time I don't rerank at all.
Having some hit rescored and others not, because of time, it's not ideal, given the fact the reRankDocs should regulate what is reranked and what not